src/views/SpecialResolution.vue

```diff
@@ -317,14 +309,35 @@ export default class SpecialResolution extends Mixins(
   /** Called when resolution summary data has changed. */
   protected async onSpecialResolutionSummaryChanges (): Promise<void> {
     // update filing data with future effective field
-    this.setFilingData({
-      ...this.getFilingData,
-      futureEffective: this.getEffectiveDateTime.isFutureEffective
+    const filingData = [...this.getFilingData]
+    filingData.forEach(fd => {
+      fd.futureEffective = this.getEffectiveDateTime.isFutureEffective
     })
+    this.setFilingData(filingData)
     // update the current fees for the filing
-    this.setCurrentFees(await this.fetchFilingFees(
-      FilingCodes.SPECIAL_RESOLUTION, this.getEntityType, this.getEffectiveDateTime.isFutureEffective
-    ))
+    await this.setCurrentFeesFromFilingData(this.getEffectiveDateTime.isFutureEffective)
+    // update the fee prices to display in the text
+    await this.setFeePricesFromFilingData(true)
+  }
+
+  /** Updates fees depending on business name change. */
+  @Watch('hasBusinessNameChanged', { immediate: true })
+  private async businessNameChanged (hasBusinessNameChanged: boolean): Promise<void> {
+    if (this.specialResolutionResource) {
+      let filingData = [this.specialResolutionResource.filingData]
+      if (hasBusinessNameChanged) {
+        filingData.push({
+          filingTypeCode: FilingCodes.SPECIAL_RESOLUTION_NAME_CHANGE,
+          entityType: CorpTypeCd.COOP,
+          priority: false
+        })
+      }
+      this.setFilingData(filingData)
+      // update the current fees for the filing
+      await this.setCurrentFeesFromFilingData(this.getEffectiveDateTime.isFutureEffective)
+      // update the fee prices to display in the text
+      await this.setFeePricesFromFilingData(true)
+    }
```

Comment:
I have a question sort of related to this. When we successfully changed the business name using name request, `hasBusinessNameChanged` will be true and the special resolution form is rendered (line 187). If you save and resume this filing, will it display the special resolution form? Another way to ask this is that did you update the `legalName` in `nameRequest`? I think this part is missing.

Reply:
Will be handled in: https://github.com/bcgov/entity/issues/13021 - Saving/Resuming a Draft Special Resolution